wsmaster/codenvy-hosted-platform-api-impl/src/test/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManagerTest.java

```diff
@@ -454,11 +471,104 @@ public void shouldPermitToCheckRamOnlyForFiveThreadsAtTheSameTime() throws Excep
-        verify(manager, timeout(300).times(5)).checkSystemRamAndPropagateStart(anyObject());
+        verify(manager, timeout(300).times(5)).checkLimitsAndPropagateStart(anyString(), anyObject());
 
         //Execute paused threads to release the throughput limit for other threads.
         invokeProcessLatch.countDown();
         //Wait for throughput limit will be released and check that RAM check was performed in other threads.
-        verify(manager, timeout(300).times(7)).checkSystemRamAndPropagateStart(anyObject());
+        verify(manager, timeout(300).times(7)).checkLimitsAndPropagateStart(anyString(), anyObject());
+    }
+
+    @Test
+    public void shouldCallCreateCallBackIfStartedWorkspacesNumberLimitIsNotExceeded() throws Exception {
+        final LimitsCheckingWorkspaceManager manager = spy(new LimitsCheckingWorkspaceManager(2,
+                                                                                              4, // max started workspaces limit
+                                                                                              "1gb",
+                                                                                              0,
+                                                                                              null,
+                                                                                              null,
+                                                                                              null,
+                                                                                              null,
+                                                                                              null,
+                                                                                              null,
+                                                                                              environmentParser,
+                                                                                              false,
+                                                                                              false,
+                                                                                              2000));
+        final WorkspaceCallback callback = mock(WorkspaceCallback.class);
+        WorkspaceImpl startingWorkspace = mock(WorkspaceImpl.class);
+        WorkspaceImpl runningWorkspace = mock(WorkspaceImpl.class);
+        WorkspaceImpl stoppingWorkspace = mock(WorkspaceImpl.class);
+        WorkspaceImpl stoppedWorkspace = mock(WorkspaceImpl.class);
+        when(startingWorkspace.getStatus()).thenReturn(WorkspaceStatus.STARTING);
+        when(runningWorkspace.getStatus()).thenReturn(WorkspaceStatus.RUNNING);
+        when(stoppingWorkspace.getStatus()).thenReturn(WorkspaceStatus.STOPPING);
+        when(stoppedWorkspace.getStatus()).thenReturn(WorkspaceStatus.STOPPED);
```

Comment:
There is a new status called `SNAPSHOTTING`, i think you should rewrite the test to be more generic, use `EnumSet` of `WorkspaceStatus` values without RUNNING status, in this way you won't need to change this test when workspace statuses are changed